src/corona_hakab_model/generation/matrix_generator.py

```diff
@@ -162,6 +163,69 @@ def create_random_clustered_circles_matrix(self, con_type: ConnectionTypes, circ
                                           p=[daily_connections_float / total_connections_float, weekly_connections_float / total_connections_float])
             v = np.full_like(conns, strengthes, dtype=np.float32)
             self.matrix[depth, agent.index, conns] = v
+            
+    def _create_geographically_clustered_circles_matrix(self, con_type: ConnectionTypes, circles: List[SocialCircle], depth):
+        # the new connections will be saved here
+        connections = [[] for _ in self.agents]
+        # gets data from matrix consts
+        connection_strength = self.matrix_consts.connection_type_to_connection_strength[con_type]
+        daily_connections_float = self.matrix_consts.daily_connections_amount_by_connection_type[con_type]
+        weekly_connections_float = self.matrix_consts.weekly_connections_amount_by_connection_type[con_type]
+        total_connections_float = daily_connections_float + weekly_connections_float
+        
+        for circle in circles:
+            agents = circle.agents
+            indexes = [agent.index for agent in agents]
+            nodes: List[Node] = [Node(index) for index in indexes]
+            
+            
+            num_agent_edges = self.matrix_consts.community_member_edges[0]
```

Comment:
as stated above, should me calculated using total_connections_float (which is the sum of daily and weekly connections).

Reply:
fixed